core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -87,6 +91,24 @@ angular.module('oppia').component('preferencesPage', {
         _saveDataItem('user_bio', userBio);
       };
 
+      ctrl.checkBioChanged = function() {
+        ctrl.allowreload = false;
+      };
+
+      ctrl.saveBeforeUnload = function() {
+        WindowRef.nativeWindow.addEventListener(
+          'beforeunload', ctrl.confirmBeforeLeaving);
+      };
```

Comment:
Any specific reason for defining this as a separate function?

```
WindowRef.nativeWindow.addEventListener(
          'beforeunload',function() {
          ....
          });
```

Reply:
Just that the function ` $q.all(` in line 246 looks neat. 

I could put the 
```
WindowRef.nativeWindow.addEventListener(
          'beforeunload', ctrl.confirmBeforeLeaving);
```
in line 256.

Reply:
I'm fine with that :) I think having an eventHandler function name will help us remove the eventListener.